File: CS/SelectQueryWindowsFormsApplication/Form1.cs
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.Sql;
using DevExpress.XtraReports.UI;
using System;
using System.Windows.Forms;
// ...

namespace SelectQueryWindowsFormsApplication {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();

            // Register the custom function.
            StDevFunction.Register();
        }

        private void button1_Click(object sender, EventArgs e) {
            // Create a data source connection.
            Access97ConnectionParameters connectionParameters = new Access97ConnectionParameters("../../nwind.mdb", "", "");
            SqlDataSource ds = new SqlDataSource(connectionParameters);

            // Create a query and specify its SELECT expression.
            SelectQuery query = SelectQueryFluentBuilder
                .AddTable("Products")
                .SelectColumn("CategoryID")
                .GroupBy("CategoryID")
                .SelectExpression("StDev([Products].[UnitPrice])", "PriceDeviation")
                .Build("Products");

            //Add the query to the data source.

            ds.Queries.Add(query);

            // Fill the data source.
            ds.Fill();

            // Create a new report and bind it to the data source.
            XtraReport report = new XtraReport();
            report.DataSource = ds;
            report.DataMember = "Products";

            // Create a report layout.
            DetailBand detailBand = new DetailBand();
            detailBand.Height = 100;
            report.Bands.Add(detailBand);

            XRLabel labelCategory = new XRLabel();
            labelCategory.ExpressionBindings.Add(
                new DevExpress.XtraReports.UI.ExpressionBinding("BeforePrint", "Text", "[CategoryID]"));
            labelCategory.TextFormatString = "Category ID: {0}";
            labelCategory.TopF = 15;
            detailBand.Controls.Add(labelCategory);

            XRLabel labelDeviation = new XRLabel();
            labelDeviation.ExpressionBindings.Add(
                new DevExpress.XtraReports.UI.ExpressionBinding("BeforePrint", "Text", "[PriceDeviation]"));
            labelDeviation.TopF = 50;
            labelDeviation.WidthF = 500;
            labelDeviation.TextFormatString = "Price Deviation: {0}";
            detailBand.Controls.Add(labelDeviation);

            // Show the report.
            report.ShowDesigner();
        }
    }
}

File: VB/SelectQueryWindowsFormsApplication/Form1.vb
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.Sql
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Windows.Forms
' ...

Namespace SelectQueryWindowsFormsApplication
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()

            ' Register the custom function.
            StDevOperator.Register()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a data source connection.
            Dim connectionParameters As New Access97ConnectionParameters("../../nwind.mdb", "", "")
            Dim ds As New SqlDataSource(connectionParameters)

            ' Create a query and specify its SELECT expression.
            Dim query As SelectQuery = SelectQueryFluentBuilder.AddTable("Products").SelectColumn("CategoryID").GroupBy("CategoryID").SelectExpression("StDev([Products].[UnitPrice])", "PriceDeviation").Build("Products")

            'Add the query to the data source.

            ds.Queries.Add(query)

            ' Fill the data source.
            ds.Fill()

            ' Create a new report and bind it to the data source.
            Dim report As New XtraReport()
            report.DataSource = ds
            report.DataMember = "Products"

            ' Create a report layout.
            Dim detailBand As New DetailBand()
            detailBand.Height = 50
            report.Bands.Add(detailBand)

            Dim labelCategory As New XRLabel()
            labelCategory.ExpressionBindings.Add(New DevExpress.XtraReports.UI.ExpressionBinding("BeforePrint", "Text", "[CategoryID]"))
            labelCategory.TextFormatString = "Category ID: {0}"

            labelCategory.TopF = 15
            detailBand.Controls.Add(labelCategory)

            Dim labelDeviation As New XRLabel()
            labelDeviation.ExpressionBindings.Add(New DevExpress.XtraReports.UI.ExpressionBinding("BeforePrint", "Text", "[PriceDeviation]"))
            labelDeviation.TopF = 30
            labelDeviation.WidthF = 500
            labelDeviation.TextFormatString = "Price Deviation: {0}"
            detailBand.Controls.Add(labelDeviation)

            ' Show the report.
            report.ShowDesigner()
        End Sub
    End Class
End Namespace
